libdnf/advisory/advisory_package_private.hpp

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright (C) 2021 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_ADVISORY_ADVISORY_PACKAGE_PRIVATE_HPP
+#define LIBDNF_ADVISORY_ADVISORY_PACKAGE_PRIVATE_HPP
+
+#include "libdnf/advisory/advisory_package.hpp"
+#include "libdnf/rpm/solv/package_private.hpp"
+#include "libdnf/rpm/solv_sack_impl.hpp"
+
+#include <solv/pooltypes.h>
+#include <solv/solvable.h>
+
+namespace libdnf::advisory {
+
+class AdvisoryPackage::Impl {
+public:
+    /// Copy constructor: clone from an existing AdvisoryPackage::Impl
+    Impl(const Impl & other);
+    /// Move constructor: clone from an existing AdvisoryPackage::Impl
+    Impl(Impl && other);
+
+    Impl & operator=(const Impl & other);
+    Impl & operator=(Impl && other);
+
+    std::string get_name() const;
+    std::string get_version() const;
+    std::string get_evr() const;
+    std::string get_arch() const;
+    Id get_name_id() const;
```

Comment:
What about to directly return the Id `Id get_name_id() const { return name; };`?